specification/app/resource-manager/Microsoft.App/preview/2023-05-01-preview/Builds.json

```diff
@@ -0,0 +1,529 @@
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "description": "The response of a BuildResource list operation.",
+      "required": [
+        "value"
+      ]
+    },
+    "BuildStatus": {
+      "type": "string",
+      "description": "Status of the build once it has been provisioned.",
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Cancelled",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "BuildStatus",
+        "modelAsString": true
+      }
+    },
+    "ContainerRegistryWithCustomImage": {
+      "type": "object",
+      "properties": {
+        "server": {
+          "type": "string",
+          "description": "Login server of the container registry that the final image should be uploaded to."
+        },
+        "identity": {
+          "type": "string",
+          "description": "Full user-assigned identity Resource ID for the managed identity that should be used to connect to the container registry."
+        },
+        "image": {
+          "type": "string",
+          "description": "Full name that the final image should be uploaded as, including both image name and tag."
+        }
+      },
+      "description": "Container registry that the final image will be uploaded to.",
+      "required": [
+        "server",
+        "identity"
+      ]
+    },
+    "EnvironmentVariable": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Environment variable name."
+        },
+        "value": {
+          "type": "string",
+          "description": "Environment variable value."
+        }
+      },
+      "description": "Model representing an environment variable.",
+      "required": [
+        "name",
+        "value"
+      ]
+    },
+    "HttpGet": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "URL to make HTTP GET request against."
+        },
+        "fileName": {
+          "type": "string",
+          "description": "Name of the file that the request should be saved to."
+        },
+        "headers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of headers to send with the request."
+        }
+      },
+      "description": "Model representing a http get request.",
+      "required": [
+        "url"
+      ]
+    },
+    "PreBuildStep": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Description of the pre-build step."
+        },
+        "scripts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of custom commands to run."
+        },
+        "httpGet": {
```

Comment:
please confirm this GET isn't made with your first party app's identity.  Also, be very careful about following redirects, allowing localhost, and other server-side request forgery type attack vectors.

Reply:
Yes I can confirm that. This GET will be made from within the container app that is identified by the `buildComputeId`. As this is created through ARM in the customer's subscription, we will not have any sensitive data / 1PA information in it. 

The request will be formatted like `curl -o {fileName} -H {header1} -H {header2} {url}`

Reply:
Ok.  They will likely be able to make requests against the IMDS endpoint that resides inside the container app.  Since this is a resource they own that likely isn't a problem but something that needs to be kept in mind.  https://learn.microsoft.com/en-us/azure/container-apps/managed-identity?tabs=cli%2Chttp#rest-endpoint-reference

Reply:
The IMDS endpoint is not available inside a Container App. SSRF is something we need to consider though; @pauld-msft as long as the Build Container App runs in the same end-user context as any other Container App AND we're not sending any sensitive values in the request (secrets, keys, tokens..) then it shouldn't be a problem (it would be no different than a user deploying their own Container App that makes HTTP requests).

Reply:
@vturecek this is part of the build configuration / pre-build stesps that the user provides. So the user would provide the values that create the http get request. Which means that we wouldn't be sending any sensitive values unless the customer decided to do so